x/concentrated-liquidity/lp.go

```diff
@@ -298,7 +302,16 @@ func (k Keeper) addToPosition(ctx sdk.Context, owner sdk.AccAddress, positionId
 		return 0, sdk.Int{}, sdk.Int{}, err
 	}
 	tokensProvided := sdk.NewCoins(sdk.NewCoin(pool.GetToken0(), amount0Desired), sdk.NewCoin(pool.GetToken1(), amount1Desired))
-	newPositionId, actualAmount0, actualAmount1, _, _, err := k.createPosition(ctx, position.PoolId, owner, tokensProvided, amount0Withdrawn, amount1Withdrawn, position.LowerTick, position.UpperTick)
+	minimumAmount0 := amount0Withdrawn
+	minimumAmount1 := amount1Withdrawn
+
+	if !amount0MinGiven.IsZero() {
+		minimumAmount0 = amount0MinGiven
```

Comment:
I was just typing this until I saw your comment. Agreed viewing this in terms of the entire amount rather than just the amount we are adding is not intuitive.

Reply:
Sorry I dont quite follow, isn't @czarcas7ic 's claim and @AlpinYukseloglu 's claim contradicting? Maybe I'm misunderstanding..

So currently the code is viewing the given (amount min given) in terms of the entire amount, where we directly use what ever given from the user as the min amount for the new position being created.

Reply:
This is breaking quite a few tests, going to make an issue for this to discuss / address separately

Reply:
https://github.com/osmosis-labs/osmosis/issues/5238